Documentation/architecture.rst

```diff
@@ -0,0 +1,155 @@
+  any L7 proxies. The socket operations hook
+  will identify candidate sockets for accelerating. These include all local node connections
+  (endpoint to endpoint) and any connection to a Cilium proxy.
+  These identified connections will then have all messages handled by the socket
+  send/recv hook and will be accelerated using sockmap fast redirects. The fast
+  redirect ensures all policies implemented in Cilium are valid for the associated
+  socket/endpoint mapping and assuming they are sends the message directly to the
+  peer socket. This is allowed because the sockmap send/recv hooks ensures the message
+  will not need to be processed by any of the objects above.
+
+* **L7 Policy:** The L7 Policy object redirect proxy traffic to a Cilium userspace
+  proxy instance. Cilium uses an Envoy instance as its userspace proxy. Envoy will
+  then either forward the traffic or drop the traffic based on the configured L7 policy.
+
+These components are connected to create the flexible and efficient datapath used
+by Cilium. Below we show the following possible flows connecting endpoints on a single
+node, ingress to an endpoint, and endpoint to egress networking device. In each case
+there is an additional diagram showing the TCP accelerated path available when Socket
+Layer Enforcement is enabled.
+
+First we show the local endpoint to endpoint flow with optional L7 Policy on
+egress and ingress. Followed by the same endpoint to endpoint flow with
+Socket layer enforcement enabled. With Socket layer enforcement enabled for TCP
+traffic the
+handshake initiating the connection will traverses the endpoint policy object until TCP state
+is ESTABLISHED. Then after the connection is ESTABLISHED only the L7 Policy
+object is still required.
+
+.. image:: /_static/cilium_bpf_endpoint.svg
+   :target: /_static/cilium_bpf_endpoint.svg
+
+Next we show local endpoint to egress with optional overlay network. In the
+optional overlay network traffic is forwarded out the Linux network interface
+corresponding to the overlay. In the default case the overlay interface is
+named cilium_vxlan. Similar to above when socket layer enforcement is enabled
+and a L7 proxy is in use we can avoid running the endpoint policy block between
+the endpoint and the L7 Policy for TCP traffic.
+
+.. image:: /_static/cilium_bpf_egress.svg
+   :target: /_static/cilium_bpf_egress.svg
+
+Finally we show ingress to local endpoint also with optional overlay network.
+Similar to above socket layer enforcement can be used to avoid a set of
+policy traversals between the proxy and the endpoint socket.
+
+.. image:: /_static/cilium_bpf_ingress.svg
+   :target: /_static/cilium_bpf_ingress.svg
+
+This complete the datapath overview. More BPF specifics can be found in the
+BPF section, :ref:`bpf_guide`. Additional details on how to extend the L7 Policy
+exist in the Envoy section, :ref:`envoy`
+
+Kubernetes Integration
```

Comment:
This section duplicates the final part of `concepts`, so we should probably remove one of them.

Reply:
that was an error I somehow lost removing it from concepts. will remove it.